jvector-base/src/main/java/io/github/jbellis/jvector/vector/VectorUtil.java

```diff
@@ -47,7 +47,7 @@ public static float dotProduct(VectorFloat<?> a, VectorFloat<?> b) {
       throw new IllegalArgumentException("vector dimensions differ: " + a.length() + "!=" + b.length());
     }
     float r = impl.dotProduct(a, b);
-    assert Float.isFinite(r);
+    assert Float.isFinite(r) : String.format("dotProduct(%s, %s) = %s", a, b, r);
```

Comment:
why doesn't this method just invoke the other with appropriate offset and length?